In [166]:
import pandas as pd
import numpy as np

#### Load the dataset as a Pandas data frame

In [21]:
VGSales_df = pd.read_csv("Data/Video_Games_Sales_as_at_22_Dcsvec_2016.csv", sep=',')

#### Display the first ten rows of data.


In [38]:
VGSales_df.head(10)

Name Platform  Year_of_Release         Genre  \
0                 Wii Sports      Wii           2006.0        Sports   
1          Super Mario Bros.      NES           1985.0      Platform   
2             Mario Kart Wii      Wii           2008.0        Racing   
3          Wii Sports Resort      Wii           2009.0        Sports   
4   Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing   
5                     Tetris       GB           1989.0        Puzzle   
6      New Super Mario Bros.       DS           2006.0      Platform   
7                   Wii Play      Wii           2006.0          Misc   
8  New Super Mario Bros. Wii      Wii           2009.0      Platform   
9                  Duck Hunt      NES           1984.0       Shooter   

  Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  \
0  Nintendo     41.36     28.96      3.77         8.45         82.53   
1  Nintendo     29.08      3.58      6.81         0.77         40.24   
2  Nintendo     15.68     12.76      3.79         3.29         35.52   
3  Nintendo     15.61     10.93      3.28         2.95         32.77   
4  Nintendo     11.27      8.89     10.22         1.00         31.37   
5  Nintendo     23.20      2.26      4.22         0.58         30.26   
6  Nintendo     11.28      9.14      6.50         2.88         29.80   
7  Nintendo     13.96      9.18      2.93         2.84         28.92   
8  Nintendo     14.44      6.94      4.70         2.24         28.32   
9  Nintendo     26.93      0.63      0.28         0.47         28.31   

   Critic_Score  Critic_Count User_Score  User_Count Developer Rating  
0          76.0          51.0          8       322.0  Nintendo      E  
1           NaN           NaN        NaN         NaN       NaN    NaN  
2          82.0          73.0        8.3       709.0  Nintendo      E  
3          80.0          73.0          8       192.0  Nintendo      E  
4           NaN           NaN        NaN         NaN       NaN    NaN  
5           NaN           NaN        NaN         NaN       NaN    NaN  
6          89.0          65.0        8.5       431.0  Nintendo      E  
7          58.0          41.0        6.6       129.0  Nintendo      E  
8          87.0          80.0        8.4       594.0  Nintendo      E  
9           NaN           NaN        NaN         NaN       NaN    NaN

#### Find the dimensions (number of rows and columns) in the data frame. What do these two numbers represent in the context of the data?

In [23]:
VGSales_df.shape

(16719, 16)

There are 16 different variables in the dataframe and 16719 records or values for each variables in the data.

#### Find the top five games by critic score.

In [93]:
VGSales_df.sort_values(by=['Critic_Score'], ascending=False)['Name'].head()

227     Tony Hawk's Pro Skater 2
57           Grand Theft Auto IV
51           Grand Theft Auto IV
5350                 SoulCalibur
165           Grand Theft Auto V
Name: Name, dtype: object

#### Find the number of video games in the data frame in each genre.

In [30]:
VGSales_df.groupby('Genre')['Name'].count()

Genre
Action          3370
Adventure       1303
Fighting         849
Misc            1750
Platform         888
Puzzle           580
Racing          1249
Role-Playing    1500
Shooter         1323
Simulation       874
Sports          2348
Strategy         683
Name: Name, dtype: int64

#### Find the first five games in the data frame on the SNES platform.

In [106]:
VGSalesSnes_df = VGSales_df[VGSales_df.Platform == "SNES"]
VGSalesSnes_df.head()
VGSalesSnes_df.sort_values(by=['Year_of_Release'], ascending=True)['Name'].head()

18      Super Mario World
1195          Final Fight
511                F-Zero
1791           Pilotwings
1111        Super Scope 6
Name: Name, dtype: object

#### Find the five publishers with the highest total global sales. Note: You will need to calculate the total global sales for each publisher to do this.

In [127]:
VGSalesPub_df = VGSales_df.groupby('Publisher').sum('Global_Sales')
VGSalesPub_df.sort_values(by=['Global_Sales'], ascending=False)['Global_Sales'].head()

Publisher
Nintendo                       1788.81
Electronic Arts                1116.96
Activision                      731.16
Sony Computer Entertainment     606.48
Ubisoft                         471.61
Name: Global_Sales, dtype: float64

#### Create a new column in the data frame that calculates the percentage of global sales from North America. Display the first five rows of the new data frame.

In [146]:
VGSales_df["NA_Percentage"] = (VGSales_df["NA_Sales"] * 100) / VGSales_df["Global_Sales"]
VGSales_df.head()

Name Platform  Year_of_Release         Genre Publisher  \
0                Wii Sports      Wii           2006.0        Sports  Nintendo   
1         Super Mario Bros.      NES           1985.0      Platform  Nintendo   
2            Mario Kart Wii      Wii           2008.0        Racing  Nintendo   
3         Wii Sports Resort      Wii           2009.0        Sports  Nintendo   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
0     41.36     28.96      3.77         8.45         82.53          76.0   
1     29.08      3.58      6.81         0.77         40.24           NaN   
2     15.68     12.76      3.79         3.29         35.52          82.0   
3     15.61     10.93      3.28         2.95         32.77          80.0   
4     11.27      8.89     10.22         1.00         31.37           NaN   

   Critic_Count User_Score  User_Count Developer Rating  NA_Percentage  
0          51.0          8       322.0  Nintendo      E      50.115110  
1           NaN        NaN         NaN       NaN    NaN      72.266402  
2          73.0        8.3       709.0  Nintendo      E      44.144144  
3          73.0          8       192.0  Nintendo      E      47.635032  
4           NaN        NaN         NaN       NaN    NaN      35.926044

#### Find the number NaN entries (missing data values) in each column.

In [147]:
VGSales_df.isna().sum()

Name                  2
Platform              0
Year_of_Release     269
Genre                 2
Publisher            54
NA_Sales              0
EU_Sales              0
JP_Sales              0
Other_Sales           0
Global_Sales          0
Critic_Score       8582
Critic_Count       8582
User_Score         6704
User_Count         9129
Developer          6623
Rating             6769
NA_Percentage         0
dtype: int64

#### Try to calculate the median user score of all the video games. You will likely run into an error because some of the user score entries are a non-numerical string that cannot be converted to a float. Find and replace this string with NaN and then calculate the median. Then, replace all NaN entries in the user score column with the median value

In [172]:
for i in VGSales_df['User_Score']:
    try:
        float(i)
    except:
        VGSales_df['User_Score'].replace(i, np.nan, inplace = True)

median_value = VGSales_df['User_Score'].median()        
VGSales_df['User_Score'] = VGSales_df['User_Score'].fillna(median_value)

VGSales_df['User_Score']   

0          8
1        7.5
2        8.3
3          8
4        7.5
        ... 
16714    7.5
16715    7.5
16716    7.5
16717    7.5
16718    7.5
Name: User_Score, Length: 16719, dtype: object